api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java

```diff
@@ -0,0 +1,263 @@
+                      .institution(inst.getInstitution())
+                      .role(inst.getRole());
+                })
+            .collect(Collectors.toList()));
+    return researcher;
+  }
+
+  private RdrWorkspace toRdrWorkspace(DbWorkspace dbWorkspace) {
+    RdrWorkspace RdrWorkspace = new RdrWorkspace();
+    RdrWorkspace.setWorkspaceId((int) dbWorkspace.getWorkspaceId());
+    RdrWorkspace.setName(dbWorkspace.getName());
+
+    RdrWorkspace.setCreationTime(dbWorkspace.getCreationTime().toLocalDateTime().atOffset(offset));
+    RdrWorkspace.setModifiedTime(
+        dbWorkspace.getLastModifiedTime().toLocalDateTime().atOffset(offset));
+    RdrWorkspace.setStatus(
+        org.pmiops.workbench.rdr.model.RdrWorkspace.StatusEnum.fromValue(
+            dbWorkspace.getWorkspaceActiveStatusEnum().toString()));
+    RdrWorkspace.setExcludeFromPublicDirectory(false);
+    RdrWorkspace.setDiseaseFocusedResearch(dbWorkspace.getDiseaseFocusedResearch());
+    RdrWorkspace.setDiseaseFocusedResearchName(dbWorkspace.getDiseaseOfFocus());
+    RdrWorkspace.setOtherPurpose(dbWorkspace.getOtherPurpose());
+    RdrWorkspace.setOtherPurposeDetails(dbWorkspace.getOtherPurposeDetails());
+    RdrWorkspace.setMethodsDevelopment(dbWorkspace.getMethodsDevelopment());
+    RdrWorkspace.setControlSet(dbWorkspace.getControlSet());
+    RdrWorkspace.setAncestry(dbWorkspace.getAncestry());
+    RdrWorkspace.setSocialBehavioral(dbWorkspace.getSocialBehavioral());
+    RdrWorkspace.setPopulationHealth(dbWorkspace.getPopulationHealth());
+    RdrWorkspace.setDrugDevelopment(dbWorkspace.getDrugDevelopment());
+    RdrWorkspace.setCommercialPurpose(dbWorkspace.getCommercialPurpose());
+    RdrWorkspace.setEducational(dbWorkspace.getEducational());
+
+    // Call Firecloud to get a list of Collaborators
+    FirecloudWorkspaceACL firecloudResponse =
+        fireCloudService.getWorkspaceAcl(
+            dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getFirecloudName());
+    Map<String, FirecloudWorkspaceAccessEntry> aclMap = firecloudResponse.getAcl();
+    aclMap.forEach(
+        (email, access) -> {
+          RdrWorkspaceUser workspaceUderMap = new RdrWorkspaceUser();
+          workspaceUderMap.setUserId((int) userDao.findUserByUsername(email).getUserId());
+          workspaceUderMap.setRole(RdrWorkspaceUser.RoleEnum.fromValue(access.getAccessLevel()));
+          RdrWorkspace.addWorkspaceUsersItem(workspaceUderMap);
+        });
+
+    return RdrWorkspace;
+  }
+
+  /**
+   * For Each entityType and entity id update lastExportDate to current date time if it exist in
+   * rdr_export table else add a new entry
+   *
+   * @param entity
+   * @param idList
+   */
+  private void updateDBRdrExport(RdrEntity entity, List<Long> idList) {
+    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
+
+    List<DbRdrExport> exportList =
+        idList.stream()
+            .map(
+                id -> {
+                  DbRdrExport rd =
```

Comment:
What does "rd" mean as a variable name? I'd lean towards using non-abbreviated variable names where possible. This could be "rdrExport".